BTProtocol/BTProtocol/BitTorrent/MainProc.cs

```diff
@@ -10,60 +10,44 @@
 using BencodeNET.Parsing;
 using BencodeNET.Torrents;
 using System.Threading;
+using System.Collections;
+using System.Threading.Tasks;
 
 namespace BTProtocol.BitTorrent
 {
-
-    enum Events : UInt16
-    {
-        started = 0,
-        paused  = 1,
-        stopped = 2,
-    }
-
-    [Serializable()]
-    struct TFData
-    {
-        public string torrent_name { get; }
-        public string resource_path { get; }
-
-        public byte[] piece_hash { get; }
-        public long piece_size { get; }
-
-        public bool[] pieces_downloaded;
-        public uint bytes_uploaded;
-        public uint bytes_downloaded;
-        public UInt64 bytes_left;
-
-
-        public Events _event;
-        public bool compact;
-
-        public TFData(string torrent_name, string resource_path, byte[] piece_hash, long piece_size)
-        {
-            this.torrent_name = torrent_name;
-            this.resource_path = resource_path;
-            this.piece_hash = piece_hash;
-            this.piece_size = piece_size;
-
-            pieces_downloaded = new bool[piece_hash.Length];
-            bytes_uploaded = 0;
-            bytes_downloaded = 0;
-            bytes_left = Convert.ToUInt64(piece_size * piece_hash.Length);
-            _event = Events.started;
-            compact = true;
-        }
-    }
-    
     class MainProc
     {
+        public static string peerid { get; set; }
```

Comment:
move to peer class,
we should probably clean up the serialized stuff and move alot of the global stuff to a separate file.

Reply:
todo in follow up pr.